File: test-project-data.App/ClientApp/src/Interfaces/IUserRepository.ts
import { User } from '../Models/User';


export interface IUserRepository {
    GetUser: (id: string) => Promise<User>,
    GetUsers: () => Promise<User[]>,
    CreateUser: (user: User) => Promise<User>
    UpdateUser: (user: User) => Promise<User>,
    DeleteUser: (id: string) => Promise<boolean>
}

File: test-project-data.App/Interfaces/IUserRepository.cs
using System;
using System.Collections.Generic;
using test_project_data.App.Models;

namespace test_project_data.App.Interfaces
{   
    public interface IUserRepository
    {
        User GetUser(Guid id);
        IEnumerable<User> GetUsers();
        User CreateUser(User user);
        void UpdateUser(User user);
        void DeleteUser(Guid id);
    }
}